src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+    metric.stddev_called_block_length = callBlockLengthCounter.stddev(m=metric.mean_called_block_length)
+    metric.n50_called_block_length    = callAssemblyStats.n50
+    metric.n90_called_block_length    = callAssemblyStats.n90
+    metric.l50_called_block_length    = callAssemblyStats.l50
+    metric.mean_truth_block_length    = truthBlockLengthCounter.mean()
+    metric.median_truth_block_length  = truthBlockLengthCounter.median()
+    metric.stddev_truth_block_length  = truthBlockLengthCounter.stddev(m=metric.mean_truth_block_length)
+    metric.n50_truth_block_length     = truthAssemblyStats.n50
+    metric.n90_truth_block_length     = truthAssemblyStats.n90
+    metric.l50_truth_block_length     = truthAssemblyStats.l50
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // Output the metrics and finish up
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    logger.info("Outputting")
+
+    Metric.write(metrics=Seq(metric), path=PathUtil.pathTo(output + AssessPhasingMetric.MetricExtension))
+    Metric.write(metrics=blockLengthMetrics, path=PathUtil.pathTo(output + PhaseBlockLengthMetric.MetricExtension))
+
+    callReader.safelyClose()
+    truthReader.safelyClose()
+  }
+
+  private def callIterator(callReader: VCFFileReader): Iterator[VariantContext] = {
+    import scala.collection.JavaConversions.asScalaIterator
+    callReader.iterator.filter(v => v.isSNP && v.isBiallelic)
+  }
+
+  private def truthIterator(truthReader: VCFFileReader, intervalList: Option[IntervalList]): Iterator[VariantContext] = {
+    (intervalList match {
+      case Some(intv) => ByIntervalListVariantContextIterator(truthReader, intv, useIndex=false)
+      case None =>
+        import scala.collection.JavaConversions.asScalaIterator
+        truthReader.iterator.toIterator
+    }).filter(v => v.isSNP && v.isBiallelic)
+  }
+}
+
+private object AssemblyStatistics {
+  def apply(counter: NumericCounter[Long]): AssemblyStatistics = {
+    if (counter.isEmpty) return AssemblyStatistics(0, 0, 0)
+
+    val numBases = counter.map { case (length, count) => length * count }.sum
+    val fiftyPercent  = numBases / 2.0
+    val ninetyPercent = numBases * 0.9
+
+    // N50/N90
+    var n50: Option[Long] = None
+    var n90: Option[Long] = None
+    var i = 0
+    var blockLengthSum = 0d
+    val countAndLengths = counter.toSeq.sortBy(_._1)
+    while (i < countAndLengths.length && (n50.isEmpty || n90.isEmpty)) {
+      blockLengthSum += countAndLengths(i)._1 * countAndLengths(i)._2
+
+      if (n50.isEmpty && fiftyPercent <= blockLengthSum) {
+        n50 = Some(countAndLengths(i)._1)
+      }
+      if (n90.isEmpty && ninetyPercent <= blockLengthSum) {
+        n90 = Some(countAndLengths(i)._1)
+      }
+
+      i += 1
+    }
+
+    // L50
+    i = countAndLengths.length - 1
```

Comment:
A few thoughts here:
1. I don't like the re-use of the `i` var .. that just seems like it'll cause errors later.  Using `forloop` would obviate that.
2. Why not just compute the L50 at the same time as the N50?
3. I think technically your L50 is not as accurate as it could be.  While for N50 reporting the length of block where the cutover happens is correct, for L50 if there are many blocks at the N50 size, you likely would only need a subset of them to get to having sum(lengths) >= 0.5 * sum(all lengths).